setup.py

```diff
@@ -32,6 +33,29 @@
 
 """
 
+def _get_AVX_macro_list():
+    cmd = """gcc -march=native -dM -E - < /dev/null | grep "__AVX512" | sort"""
```

Comment:
what does this command do?

Reply:
The part of「gcc -march=native -dM -E - < /dev/null」
generates a list of ‘#define’ directives for all the macros defined during the execution of the preprocessor.
  c.f. https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html#Preprocessor-Options

And, 「grep "__AVX512"」 extracts AVX512 macro.

Reply:
this feels a bit hacky – i'm nervous this won't work on all platforms (windows definitely not)

Reply:
So, that's it. 
In windows gcc doesn't exists, so this code can't work.
sorry, I rethink.